clients/src/main/java/org/apache/kafka/clients/consumer/internals/metrics/KafkaConsumerMetrics.java

```diff
@@ -91,16 +92,20 @@ public KafkaConsumerMetrics(Metrics metrics, String metricGrpPrefix) {
     }
 
     public void recordPollStart(long pollStartMs) {
+        if (this.pollEndMs != 0) {
+            long pollIntervalMs = pollStartMs - this.pollStartMs;
+            long pollTimeMs = this.pollEndMs - this.pollStartMs;
+            double pollIdleRatio = pollTimeMs * 1.0 / pollIntervalMs;
+            this.pollIdleSensor.record(pollIdleRatio);
+        }
         this.pollStartMs = pollStartMs;
         this.timeSinceLastPollMs = lastPollMs != 0L ? pollStartMs - lastPollMs : 0;
         this.timeBetweenPollSensor.record(timeSinceLastPollMs);
         this.lastPollMs = pollStartMs;
     }
 
     public void recordPollEnd(long pollEndMs) {
-        long pollTimeMs = pollEndMs - pollStartMs;
-        double pollIdleRatio = pollTimeMs * 1.0 / (pollTimeMs + timeSinceLastPollMs);
-        this.pollIdleSensor.record(pollIdleRatio);
+        this.pollEndMs = pollEndMs;
```

Comment:
I think this wouldn't record a metric when there's only one invocation of `poll()`. This is also evident from the failing `KafkaConsumerTest::testPollIdleRatio` test.